staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1460,6 +1460,16 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 				createdPods = append(createdPods, out)
 			}
 
+			err := wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, wait.ForeverTestTimeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
won't harm in general

Reply:
We can be smarter here. What you want is just a single List request setting `NotOlderThan` semantic with RV set to last pod creation RV.

That underneath ensures watchcache is synced to that RV.

Reply:
I like getting smarter. Thanks!

Reply:
pushed, ptal.